packages/vite/src/node/server/ws.ts

```diff
@@ -43,7 +44,7 @@ export function createWebSocketServer(
     })
   } else {
     const websocketServerOptions: ServerOptions = {}
-    const port = (hmr && hmr.port) || 24678
+    const port = hmrPort || 24678
```

Comment:
(cont https://github.com/vitejs/vite/pull/7318/files#r831745457)

…then this line can be removed entirely